src/main/java/com/palantir/gradle/versions/GradleConfigurations.java

```diff
@@ -36,14 +35,18 @@
     /**
      * Filters out both the unresolvable configurations but also the legacy java configurations that should not be
      * resolved.
+     *
+     * Note that we need to do a defensive copy here to guard against concurrent modification.
+     * See https://github.com/palantir/gradle-consistent-versions/pull/812
      */
-    public static Stream<Configuration> getResolvableConfigurations(Project project) {
+    public static Set<Configuration> getResolvableConfigurations(Project project) {
```

Comment:
👍  I've always found APIs which return `Stream` a bit of a code-smell. It's not quite reactive enough for full benefits (no backpressure mechanism) but introduces similar layers of complexity when the stream can be lazily evaluated.